tests/unit/decorators_test.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+from pydeepmerge import deep_merge
+from pydeepmerge.strategies.decorators import (
+    recurse,
+    keep_default,
+    disallow_mixed,
+)
+
+
+def return_right(left, right):
+    return right
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "test_input,expected",
+    [
+        (
+            (
+                {"a": 0, "b": {"a": 0, "b": {"a": 0, "b": 0}}},
+                {"b": {"b": {"b": 1}}},
+            ),
+            {'a': 0, 'b': {'a': 0, 'b': {'a': 0, 'b': 1}}}
+        ),
+    ]
+)
+def test_recurse(test_input, expected):
```

Comment:
nitpick: these tests could all be collapsed under a single test function.

Adding as #5 so it can be tackled some other time